### PR TITLE
fix/elfwriter:wrong first program header type null

### PR DESCRIPTION
**Pull request by zdyj3170101136**

### CODE REVIEW

File: pkg/elfwriter/elfwriter.go [hunk 1]
```diff
@@ -630,9 +630,9 @@ func (w *Writer) writeSections() {
 	w.seek(w.seekSectionHeader, io.SeekStart)
 	w.u64(uint64(shoff))
 	w.seek(w.seekSectionNum, io.SeekStart)
-	w.u64(uint64(shnum)) // e_shnum
+	w.u16(uint16(shnum)) // e_shnum
```
Comment (kakkoyun) [inline pkg/elfwriter/elfwriter.go:633]: Could you please point me to the section of the documentation/specification for the ELF?

**Review by kakkoyun — COMMENTED** [review]
Looking good.
To make a sanity check, could you add related references to the specification to the PR description?

File: pkg/elfwriter/elfwriter.go (hunk 1 above)
Comment (zdyj3170101136) [reply pkg/elfwriter/elfwriter.go:633]: there is already have sectionHeader struct in function elfwriter.writeFileHeader.
so it is no need to add duplicate comment.
截屏2023-01-11 下午6 41 15

Comment (kakkoyun) [reply pkg/elfwriter/elfwriter.go:633]: Aha ok, apologies my mistake 🤗

**Review by kakkoyun — APPROVED** [review]
LGTM

**Pull request merged**